src/Microsoft.Identity.Web.OWIN/OpenIdConnectCachingSecurityTokenProvider.cs

```diff
@@ -76,12 +76,12 @@ public IEnumerable<SecurityKey>? SecurityKeys
             }
         }
 
-        private void RetrieveMetadata()
+        private async Task RetrieveMetadataAsync()
         {
             _synclock.EnterWriteLock();
             try
             {
-                OpenIdConnectConfiguration config = Task.Run(_configManager.GetConfigurationAsync).Result;
+                OpenIdConnectConfiguration config = await _configManager.GetConfigurationAsync();
```

Comment:
Need to revert and block it here.

https://stackoverflow.com/questions/59048102/c-sharp-fire-and-forget-task-and-discard

Reply:
I'm a little confused, does `await` not block until the result of the `Task` is needed?

Reply:
`await` doesn't block ("block" has a specific meaning here, `.Result` blocks), but it suspends the method and returns the control to the calling method, which since it doesn't await the result, just continues executing synchronously. So not awaiting ends up making this a fire-and-forget operation.

See:
- Step 6 in [What happens in an async method](https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/task-asynchronous-programming-model#BKMK_WhatHappensUnderstandinganAsyncMethod) 
- [async and await](https://learn.microsoft.com/en-us/dotnet/csharp/asynchronous-programming/task-asynchronous-programming-model#BKMK_AsyncandAwait)
- More detail: [How Async/Await Really Works in C#](https://devblogs.microsoft.com/dotnet/how-async-await-really-works/)

Reply:
Understood. I will revert